### PR TITLE
Cyborg Campaign

### PR DESCRIPTION
**Pull request by Kert**
- Fixed crashes related to starting campaign games
- Cyborg Campaign is fully functional albeit with dummy metadata. Hover over right bottom corner in expansion campaign selection screen to play it
- All campaigns (original and expansion ones) now support ironfist save xml format
- Added support for playing non-expansion campaigns without CD
- More descriptive names for various fields of game class

### CODE REVIEW

File: src/cpp/shared/campaign/campaign.cpp [hunk 1]
```diff
@@ -92,4 +295,108 @@ int ExpCampaign::HandleVictory() {
...
+      case 8u:
+        PlaySmacker(SMACKER_XCAMPAIGN_DESCENDANTS_6);
+        break;
+      default:
+        return;
+    }
+  }
+}
+
+void ExpCampaign::ShowInfo(int inGame, int unused) {
```
[the remark this answers is not in the document]
Comment (Kert) [reply src/cpp/shared/campaign/campaign.cpp]: Shows campaign info where you choose bonuses before starting the map. And also does the same thing in-game when you press on scenario info

File: src/cpp/shared/init.cpp [hunk 2]
```diff
@@ -148,7 +149,10 @@ void __fastcall SetupCDRom() {
 }
 
 int __fastcall SetupCDDrive() {
-	return 1;
+  int res = SetupCDDrive_orig();
+  if(res)
+    return 1;
+  return res;
```
[the remark this answers is not in the document]
Comment (Kert) [reply src/cpp/shared/init.cpp]: I guess this should be removed all together. I think I was bit confused when starting any campaign crashed the game but the cause was different. Just tested that having just original SetupCDDrive acts in the same way as in this branch

**Comment by jkoppel** [comment]
I've confirmed that CD mode is again working, but it's still not fading properly when I complete a mission.

**Comment by Kert** [comment]
> I've confirmed that CD mode is again working, but it's still not fading properly when I complete a mission.

Can't replicate. Which missions are you completing? Is this during no-cd mode?

**Comment by jkoppel** [comment]
Scenario 0, completed without  the ISO mounted

**Comment by Kert** [comment]
> Scenario 0, completed without the ISO mounted
Scenario 0 is basically Cyborg campaign? Can't replicate, it works fine for me

**Pull request merged**